src/framework/annotator_mgr.cpp

```diff
@@ -583,7 +576,185 @@ namespace uima {
+          }
+        } catch (Exception &exception) {
+          if (!frame.originalFlow->continueOnFailure(frame.lastEngineKey /* ,exception */))
+            throw;
+        }
+
+        if (!currentCas) {
+          // if there is no more output CASes from the stack, take the original CAS that was processed by
+          // the CAS Multiplier and continue with its flow
+          currentCas = frame.originalCas;
+          flow = std::move(frame.originalFlow);
+          currentCas->setCurrentComponentInfo(nullptr);
+          casIterStack.pop();
+        }
+
+        activeCASes.insert(currentCas);
+
+        if (nextStep.getType() == Step::StepType::UNSPECIFIED) {
+          nextStep = flow->next(); // get the next step for the current flow
+        }
+
+        while (nextStep.getType() != Step::StepType::FINALSTEP) {
+          if (nextStep.getType() == Step::StepType::SIMPLESTEP) {
+            // find the AE specified by the step
+            const icu::UnicodeString &nextAEKey = nextStep.getSimpleStep()->getEngineName();
+            auto it = std::find_if(iv_vecEntries.begin(), iv_vecEntries.end(),
+                                   [&, nextAEKey](const EngineEntry &entry) {
+                                     return entry.iv_pEngine->getAnnotatorContext().iv_AnCKey == nextAEKey;
+                                   });
+
+            if (it != iv_vecEntries.end()) {
+              AnalysisEngine *nextAE = it->iv_pEngine;
+              CAS *outputCas = nullptr;
+
+              // call process one the AE and see if it has produced a new CAS
+              try {
+                CASIterator casIter = nextAE->processAndOutputNewCASes(*currentCas);
+                if (casIter.hasNext())
+                  outputCas = &casIter.next();
+              } catch (Exception &e) {
+                if (!flow->continueOnFailure(nextAEKey))
+                  throw;
+              }
+
+              if (outputCas) {
+                // new CAS is output so put the current components on the stack so we can process
+                // the other output CASes and original CASes later
+                std::unique_ptr<Flow> nextFlow = flow->newCasProduced(*outputCas, nextAEKey);
+                casIterStack.push({nextAE, currentCas, std::move(flow), nextAEKey});
+                flow = std::move(nextFlow);
+                currentCas = outputCas;
+                activeCASes.insert(currentCas);
+              } else {
+                // No new CASes are output, this CAS is done being processed by the current engine.
+                currentCas->setCurrentComponentInfo(nullptr);
+              }
+            } else {
+              UIMA_EXC_THROW_NEW(EngineProcessingException,
+                                 UIMA_ERR_USER_ANNOTATOR_COULD_NOT_PROCESS,
+                                 UIMA_MSG_ID_EXCON_PROCESSING_CAS,
+                                 ErrorMessage(UIMA_MSG_ID_LITERAL_STRING, "Unknown Delegate Key"),
```

Comment:
any chances of including the key itself in the error message?

Reply:
Done